author_postprocessing/author_postprocessing.py

```diff
@@ -0,0 +1,233 @@
+# coding=utf-8
+# This file is part of codeface-extraction, which is free software: you
+# can redistribute it and/or modify it under the terms of the GNU General
+# Public License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2015-2017 by Claus Hunsen <hunsen@fim.uni-passau.de>
+# Copyright 2020 by Thomas Bock <bockthom@cs.uni-saarland.de>
+# All Rights Reserved.
+"""
+This file is able to disambiguate authors after the extraction from the Codeface database was performed. A manually
+created disambiguation file is used to disambiguate the authors in all the extracted files of a project.
+
+The manually created disambiguation file 'disambiguation-after-db.list' has to have the following format:
+    - each line combines two person identities which should be mapped to each other
+    - each line consists of six columns (each three for describing id, name, e-mail address)
+    - the entries of each line are taken from the global 'authors.list' file
+    - Example:
+        1234;Claus Hunsen;claus.hunsen@example.org;5678;claushunsen;claushunsen@hunsen.org;
+    - the first three columns of a line describe the person identity to keep (e.g., 1234)
+    - the last three columns of a line describe the person identity to replace (e.g., 5678)
+Result: Every occurrence of the second person identity will be replaced by the first person identity, in every .list
+file of the project (authors, commits, emails, issues, etc.)
```

Comment:
> Is there any backup of the existing files?

Not yet. I just thought that as soon as we have run this new disambiguation step, nobody will be interested in the data from the extraction itself anymore. (And I won't apply this new step to the previous results' directories. So, it only will affect new data extractions.)

> If not, I would like to have this possibility (as a CLI switch, for example). For some projects, it really takes some time to re-extract the data from the database and having backup files may become handy.

A CLI switch sounds like a good idea (better than copying all results and wasting disk space per default). Most extractions are very fast, but you're right, there are some extractions which take some time.

> E.g., move all adjusted files to a timestamped backup folder while respecting their relative paths. thinking What do you think?

Hm. I'm not sure whether having timestamped backup folders is useful. This would create new backup folders whenever the extraction is run, resulting in many confusing backup folders. If you would like to have such, you can just run a backup script on the complete data. So, it is not necessary to integrate this into `codeface-extraction`.

When backuping the data by `codeface-extraction`, then I only have two ideas:
1. just copy and move the changed files, e.g., copy `authors.list` to `authors.bak.list` (or any other comprehensible name), for each of the affected files
2. copy the complete directory (e.g., `proximity` to `proximity.bak`) and only copy the `.list` files (to get rid of all the other cumbersome files created by `codeface`).

Both options, IMHO, fulfill all the requirements for such a backup. The second option also easily allows to integrate anonymization scripts in a similar way later on. So, the second option sounds more extensible and reusible to me (e.g., having a separate `proximity.anonymized` folder later -- but this are just dreams of the future, not to be discussed in this PR).

Reply:
> > E.g., move all adjusted files to a timestamped backup folder while respecting their relative paths. thinking What do you think?
> 
> Hm. I'm not sure whether having timestamped backup folders is useful. This would create new backup folders whenever the extraction is run, resulting in many confusing backup folders. If you would like to have such, you can just run a backup script on the complete data. So, it is not necessary to integrate this into `codeface-extraction`.
> 
> When backuping the data by `codeface-extraction`, then I only have two ideas:
> 
>     1. just copy and move the changed files, e.g., copy `authors.list` to `authors.bak.list` (or any other comprehensible name), for each of the affected files
> 
>     2. copy the complete directory (e.g., `proximity` to `proximity.bak`) and only copy the `.list` files (to get rid of all the other cumbersome files created by `codeface`).
> 
> 
> Both options, IMHO, fulfill all the requirements for such a backup. The second option also easily allows to integrate anonymization scripts in a similar way later on. So, the second option sounds more extensible and reusible to me (e.g., having a separate `proximity.anonymized` folder later -- but this are just dreams of the future, not to be discussed in this PR).

I thougth about the timestamped folders as repeated backups may overwrite existing backups and, thus, the original files. Apart from that, both ideas are good proposals, with the folder-based one being the "easier" one.

Reply:
> I thougth about the timestamped folders as repeated backups may overwrite existing backups and, thus, the original files. 

That's true, good point! However, I still dislike the timestamped folders because of the additional maintenance effort to remove outdated folders and save disk space.

One option to mitigate the overwrite issue could be to perform the backup only if it is not present yet. That is, if the backup directory exists, don't perform the backup up if the backup flag is set. Nevertheless, this option also has its drawbacks: Even if the extraction from the database is run anew, the backup wouldn't be performed anew if the backup folder exists already. Either one has to remove the directory manually whenever a new backup should be created - or we automatically remove the backup folder whenever the extraction is run anew, but automatically removing backups is also a little bit scary - so I would prefer the manual effort of removing the backup folder (it could also be done automatically in external bash scripts but not within the extraction itself).

> Apart from that, both ideas are good proposals, with the folder-based one being the "easier" one.

Yeah, I also talked to @hechtlC and we agreed on the folder-based one. This also eases access to the backup: In coronet, for example, you only have to change one directory name to access the backuped data. So, I will implement that.

Reply:
I have implemented the backup functionality in e9ab19a866edae7740f293103f0e2a40b87d0701, as suggested above.
@clhunsen please check if my implementation fulfils your requirements. Thanks!